tests/decorators/test_python.py

```diff
@@ -506,6 +506,62 @@ def add_2(number: int):
 
         assert ret.operator.doc_md.strip(), "Adds 2 to number."
 
+    def test_user_provided_task_id_in_a_loop_is_used(self):
+        """Tests that when looping that user provided task_id is used"""
+
+        @task_decorator(task_id='hello_task')
+        def hello():
+            """
+            Print Hello world
+            """
+            print("Hello world")
+
+        with self.dag:
+            for i in range(3):
+                hello.override(task_id=f'my_task_id_{i * 2}')()
```

Comment:
Let’s add a `hello()` after the loop to ensure `hello_task` is still usable (not mutated) after `override` is called.